frontend/src/js/components/PageViewer/PageCache.ts

```diff
@@ -0,0 +1,100 @@
+import authFetch from "../../util/auth/authFetch";
+import { CachedPreview, Page } from "./model";
+import { renderPdfPreview } from "./PdfHelpers";
+
+export class PageCache {
+  uri: string;
+
+  static MAX_CACHED_PAGES = 100;
+
+  // Tracking of text (Page}) and rendered previews (HTMLCanvasElement) is handled seperately incase an
+  // eviction happens to an item that is in flight
+  previews: CachedPreview[] = [];
```

Comment:
why is this type called `CachedPreview` and the other `Page` (not `CachedPage`, or `CachedPageData`)?

I reckon they should be called `PagePreview` and `PageData` - there's nothing intrinsic in the type that means it's definitely cached, so could be misleading otherwise

Reply:
`CachedPreview` contains information which is relevant to how the data is going to be accessed in practice by the systems that rely on the cache. Specifically the rendered canvas, the scale used, and the `PDFPageProxy`.

It's named like this because this collection of fields only makes sense in the context of working with cached results, where we want to minimise computation.

Reply:
`CachedPreview` is named like that because it contains data pertinent to consumers of the cache, in this case the rendered canvas (the "Preview" in this case), the scale, and the raw `PDFPageProxy`. 

Consumers of the cache need to consume these 3 elements but for different things, in no particularly coherent or abstract way so the type is a bit tricky to name... `CachedPreviewResult` is the only thing I can think of but that's not really adding anything... idk